mmf/modules/encoders.py

```diff
@@ -103,9 +148,27 @@ def forward(self, image):
         return self.module(image)
 
 
+@dataclass
+class ResNet152ImageEncoderParams(EncoderParams):
+    pretrained: bool = True
+    # "avg" or "adaptive"
+    pool_type: str = "avg"
+    num_output_features: int = 1
+
+
+@dataclass
+class ResNet152ImageEncoderConfig(ImageEncoderConfig):
```

Comment:
Do you think it is better to make the 'Config' classes as nested inside the main class? For example `ResNet152ImageEncoderConfig` can be a nested class inside `ResNet152ImageEncoder`.

Reply:
This is a very good point for discussion. I was thinking about this based on what PyText does. Then, on the other hand, this is style in which transformer does it. Let me think a little more about this. I do agree that keeping Config objects inside the actual module using it is better as it keep it grouped, but on the other side it also increases some friction in reusability.